### PR TITLE
all-libary: add voronoi support and git version

### PR DESCRIPTION
**Pull request by junghans**

### CODE REVIEW

**Comment by junghans** [comment]
@spack/merge-to-develop please review!

File: var/spack/repos/builtin/packages/all-library/package.py [hunk 1]
```diff
@@ -15,21 +15,25 @@ class AllLibrary(CMakePackage):
 
     homepage = "http://slms.pages.jsc.fz-juelich.de/websites/all-website/"
     url = "https://gitlab.jsc.fz-juelich.de/SLMS/loadbalancing/-/archive/v0.9.2/loadbalancing-v0.9.2.tar.gz"
+    git = "https://gitlab.jsc.fz-juelich.de/SLMS/loadbalancing"
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/all-library/package.py]:
```
$ git clone https://gitlab.jsc.fz-juelich.de/SLMS/loadbalancing
Cloning into 'loadbalancing'...
warning: redirecting to https://gitlab.jsc.fz-juelich.de/SLMS/loadbalancing.git/
```
so:
```suggestion
    git = "https://gitlab.jsc.fz-juelich.de/SLMS/loadbalancing.git"
```

File: var/spack/repos/builtin/packages/all-library/package.py [hunk 2]
```diff
@@ -15,21 +15,25 @@ class AllLibrary(CMakePackage):
...
 
     license("BSD-3-Clause", checked_by="junghans")
 
+    version("master", branch="master")
     version("0.9.2", sha256="2b4ef52c604c3c0c467712d0912a33c82177610b67edc14df1e034779c6ddb71")
 
     variant("fortran", default=False, description="Build with fortran support")
     variant("shared", default=True, description="Build shared libraries")
     variant("vtk", default=False, description="Build with vtk support")
+    variant("voronoi", default=False, description="Enable voronoi-based loadbalancing scheme")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/all-library/package.py]:
```suggestion
    variant("voronoi", default=False, description="Enable voronoi-based loadbalancing scheme", when="@0.9.3:")
```

File: var/spack/repos/builtin/packages/all-library/package.py [hunk 3]
```diff
@@ -15,21 +15,25 @@ class AllLibrary(CMakePackage):
...
 
     depends_on("c", type="build")
     depends_on("cxx", type="build")
     depends_on("fortran", type="build", when="+fortran")
     depends_on("vtk", when="+vtk")
+    depends_on("voropp", when="+voronoi @0.9.3:")
```
Comment (wdconinc) [inline var/spack/repos/builtin/packages/all-library/package.py]:
```suggestion
    depends_on("voropp", when="+voronoi")
```

**Review by wdconinc — COMMENTED** [review]
It's the variant that is only available as of 0.9.3, not that the dependency only applies as of 0.9.3.

**Comment by junghans** [comment]
@wdconinc thanks!

**Review by wdconinc — DISMISSED** [review]
Looks good now, pending @spackbot fix style

**Comment by junghans** [comment]
@spackbot fix style

**Review by wdconinc — APPROVED** [review]

**Pull request merged**